data/transactions/logic/eval.go

```diff
@@ -3422,8 +3447,8 @@ func opTxBegin(cx *EvalContext) {
 	fee := cx.Proto.MinTxnFee
 	if cx.FeeCredit != nil {
 		// Use credit to shrink the fee, but don't change FeeCredit
-		// here, because they might never tx_submit, or they might
-		// change the fee.  Do it in tx_submit.
+		// here, because they might never itxn_submit, or they might
+		// change the fee.  Do it in itxn_submit.
 		fee = basics.SubSaturate(fee, *cx.FeeCredit)
 	}
 	cx.subtxn.Txn.Header = transactions.Header{
```

Comment:
Hi, a quick question. Should we assume the inner tx belongs to the same group/lease than the parent despite not present and should be added when calculating the ID of the inner tx?

Reply:
The inner txn has no id.

Reply:
Ahh ok, thanks for the info because, from the indexers and from AlgoExplorer point of view, we didn't know if these txs could be identified individually.

Reply:
Indexer may well put a fake id in its internal table in order to place itxns in the same table with top-level.  But you would obtain the info pertaining to an itxn only by asking for the top-level, and receive a tree back.

Reply:
Yes, that's a good idea and will see with @winder the approach to show the same fake id.

Reply:
Using fake IDs may allow Indexer/AlgoExplorer users to search for an inner transaction easily using a regular TXID.

If we do that, we should agree on a definition for such fake txids. Maybe it could be something like `digest(tx, parent, offset)`

Reply:
I don't love the fake txid thing. I hope we'll keep it an implementation detail and avoid confusing users.

Reply:
And what about adding an "inner transaction index" field to the subtxn. Node must check this field is not defined or zero when a TX is submitted. This plus the first/last round you already added, should do the work.